### PR TITLE
Use QLiteHtml for preview (experimental)

### PR DESCRIPTION
**Pull request by Waqar144**
This change introduces QLiteHtml for previewing. QLiteHtml brings with it full Html/Css support so it is able to render the preview a lot better than what we currently use.

Right now it is only availble if you build QOwnNotes yourself with following enabled:

### using cmake
```cmake
cmake -DUSE_QLITEHTML=ON ..
```
### using qmake
```
qmake CONFIG+=USE_QLITEHTML ../
```

I have tested things and I havent' found anything broken really. The selection is not too good, but useable. There is no regexp search, but that can be implemented I think. Rest everything seems to work as expected.

Also, I think it will be hard to port to older Qt versions and by default support Qt >= 5.11. Thus keeping both solutions is the only way if we want to keep old Qts.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -40,6 +40,12 @@ endif()
 add_subdirectory(libraries/fakevim)
 add_subdirectory(libraries/diff_match_patch)
 
+option(USE_QLITE_HTML "Build using system installed Botan 2 crypto library" OFF)
```
Comment (pbek) [inline src/CMakeLists.txt:43]: What has USE_QLITE_HTML to do with Botan?

Comment (Waqar144) [reply src/CMakeLists.txt:43]: Copy paste ;)

Comment (pbek) [reply src/CMakeLists.txt:43]: I'll change that to "QLiteHtml" then... 😉

**Comment by pbek** [comment]
So far I just had some styling issues with this markdown:

```markdown
*Italic*
_italic_

**bold**

__underline__
```

**Comment by pbek** [comment]
Underline seems to have more priority what we give it in our note edit:

image

**Comment by pbek** [comment]
Other things I found that didn't work:

- checkbox lists
- context menu
- yes, selections still feel strange 😁 

image

**Comment by pbek** [comment]
But the Qt 5.9 build is broken! https://github.com/pbek/QOwnNotes/runs/4676252035?check_suite_focus=true

**Comment by pbek** [comment]
Your singleshot: https://github.com/pbek/QOwnNotes/runs/4676252035?check_suite_focus=true#step:14:1575

**Comment by Waqar144** [comment]
Context menu support will need to be added manually, but that should be fairly easy to add

**Comment by Waqar144** [comment]
Proper checkbox support also need to be added, but the one we were using previously works if I recall correctly.

**Comment by Waqar144** [comment]
> Underline seems to have more priority what we give it in our note edit:
> 
> image

This seems to work as expected. Am I missing something? The italic in the editor maybe a style just, and in fact shouldn't be italic + underline. The preview is correct in rendering it just as italic.

**Comment by pbek** [comment]
> This seems to work as expected. Am I missing something? 

Ah, the old preview styles it the same way. My bad...

**Comment by pbek** [comment]
Ok, I'm building a new release (without flags! 😁). Let's see if anything breaks for older versions of Qt. 😅

**Comment by Waqar144** [comment]
I am interested in whether the fancy markdown styling css stylesheets will work correctly with this. Qlitehtml boasts that it can render bootstrap correctly, so seems like it should work (assuming the actual painting is being done correctly).

**Comment by pbek** [comment]
...all the possibilities! 😁

**Comment by pbek** [comment]
No building issues so far with QLiteHtml turned off...

**Comment by Waqar144** [comment]
> No building issues so far with QLiteHtml turned off...

Great!

**Pull request merged**